src/lib/homestore/homeobject.cpp

```diff
@@ -20,4 +25,27 @@ void HomeObjectImpl::init_repl_svc() {
     }
 }
 
+HSHomeObject::HSHomeObject(std::weak_ptr< HomeObjectApplication >&& application) : HomeObjectImpl(std::move(application)) {
+    //register some callbacks for metadata recovery;
+    homestore::hs()->meta_service().register_handler(HomeObjectImpl::s_shard_info_sub_type,
```

Comment:
I plan on having an `init_homestore` in my other "Initialize HomeStore" PR since we'll likely have to hand it a weak/shared ptr to ourself for callbacks and can't happen in the constructor. We can move this there and keep the `using` default constructor after that PR merges.

Reply:
sure, will do that after init homestore pr is merged.